### PR TITLE
enable silent compare of parameter

### PR DESCRIPTION
**Pull request by BazookaJoe1900**
**Describe problem solved by the proposed pull request**
Remove false errors after comparing parameters that doesn't exists.
as described in https://github.com/PX4/Firmware/issues/12832

**Test data / coverage**
This is draft, to look

**Describe your preferred solution**
new compare-silent, greater-silent commands. that should be used in cases that we know that there is possibility that the parameter doesn't exist at all.

### CODE REVIEW

**Comment by dagar** [comment]
A quieter init would be nice, but I also don't want to completely lose track of irrelevant cruft. Would it make sense to change the default to silent, but have some mechanism to enable the error output (build time or environment variable)?

**Comment by dagar** [comment]
It also makes me wonder if we could do something more fundamental to capture a portion of the straightforward cases. 

Examples:
https://github.com/PX4/Firmware/blob/f0ee0b5d495224cdc517c216bd304094df9f0817/ROMFS/px4fmu_common/init.d/rcS#L439-L444

https://github.com/PX4/Firmware/blob/f0ee0b5d495224cdc517c216bd304094df9f0817/ROMFS/px4fmu_common/init.d/rcS#L446-L450

https://github.com/PX4/Firmware/blob/f0ee0b5d495224cdc517c216bd304094df9f0817/ROMFS/px4fmu_common/init.d/rcS#L452-L456

Why don't we just generate that at build time if the module is part of that particular config?

**Comment by BazookaJoe1900** [comment]
@dagar about the first comment, I think that the default shoud be non-silent
about generating RCs according to build, If you can show me how that can be done I can start to implement that

**Comment by BazookaJoe1900** [comment]
@julianoes I agree about --silent' but to '-s'. I will do the change later.
BTW, I think that script should not even try to call parameter that doesn't exists. but current implementation doesn't fit to that.
I have contacted with @dagar about that and talked about some kind of manifest that will generate rcS parts according to spacific compilation modules.

**Comment by dagar** [comment]
The `-s` seems fine, although I'm still wondering where we actually care about keeping the non-silent version.

**Comment by BazookaJoe1900** [comment]
I think that the non-silent version should be the default. if we could we would have make it a build time error. But because we can't... only in cases we know we might "break" the rule of not trying to get a non-exist parameter we will use the silent compare.
I can think of a method that during build time I can give a warrning/error if a RC script try to fetch non-exist parameter

File: src/systemcmds/param/param.cpp [hunk 1]
```diff
@@ -70,6 +70,12 @@ enum COMPARE_OPERATOR {
 	COMPARE_OPERATOR_GREATER = 1,
 };
 
+enum COMPARE_ERROR_LEVEL {
```
Comment (dagar) [inline src/systemcmds/param/param.cpp]: enum class

**Comment by BazookaJoe1900** [comment]
@julianoes Thanks!
Now the real "test" is to verify that there are no Errors running the startup script. that is something that I know that @dagar working on.
You delete the branch on the PX4/Firmware/pr-param_compare_silent

**Pull request merged**